### PR TITLE
feat: Add November 29th spell 

### PR DESCRIPTION
**Pull request by lucas-manuel**

### CODE REVIEW

**Review by barrutko — COMMENTED** [review]
I confirm that the only change that is made is change of the `supplySpread` from `0` to `5000000000000000000000000`, which is equal to the `borrowSpread`

**Note:** Wrong directory for the spell. It should be `20231129`, not `20230913`

File: diffs/20230913-Ethereum-0xC13e21B648A5Ee794902342038FF3aDAB66BE987-pre_20230913-Ethereum-0xC13e21B648A5Ee794902342038FF3aDAB66BE987-post.md [hunk 1]
```diff
@@ -1,38 +1,5 @@
-## Reserve changes
```
Comment (hexonaut) [inline diffs/20230913-Ethereum-0xC13e21B648A5Ee794902342038FF3aDAB66BE987-pre_20230913-Ethereum-0xC13e21B648A5Ee794902342038FF3aDAB66BE987-post.md:1]: This should not be getting modified. Probably from the directory being named incorrectly.

File: src/proposals/20230913/SparkEthereum_20231129.sol [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.10;
+
+import { SparkPayloadEthereum } from '../../SparkPayloadEthereum.sol';
+
+/**
+ * @title  November 29, 2023 Spark Ethereum Proposal - Set DAI supply spread to 0.5%
+ * @author Phoenix Labs
+ * @dev    This proposal updates DAI Interest Rate Strategy supplySpread parameter to match borrowSpread parameter.
+ * Forum:           TODO
```
Comment (hexonaut) [inline src/proposals/20230913/SparkEthereum_20231129.sol:10]: Forum post is up: https://forum.makerdao.com/t/accounting-discrepancy-in-the-dai-market/22845

File: src/proposals/20230913/SparkEthereum_20231129.sol [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * DAI Vote:        TODO
+ * Flash Loan Vote: TODO
```
Comment (hexonaut) [inline src/proposals/20230913/SparkEthereum_20231129.sol:12]: Don't need these.

File: src/proposals/20230913/SparkEthereum_20231129.t.sol [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        ( , uint256 accruedToTreasuryScaled,,,,,,,,,, ) = dataProvider.getReserveData(asset);
+
+        ( address aToken,, ) = dataProvider.getReserveTokensAddresses(asset);
+
+        uint256 totalDebt      = dataProvider.getTotalDebt(asset);
+        uint256 totalLiquidity = IERC20(asset).balanceOf(aToken);
+        uint256 totalDeposits  = IERC20(aToken).totalSupply();
+
+        uint256 assets      = totalLiquidity + totalDebt;
+        uint256 liabilities = totalDeposits + accruedToTreasuryScaled;
```
Comment (hexonaut) [inline src/proposals/20230913/SparkEthereum_20231129.t.sol:151]: This should be `(IERC20(aToken).scaledTotalSupply() + accruedToTreasuryScaled) * liquidityIndex` although for DAI there is no reserve factor so treasury amount will be 0 anyways.

**Review by hexonaut — CHANGES_REQUESTED** [review]
Couple of minor issues. Rest looks good. Time series test looks solid.

File: src/proposals/20230913/SparkEthereum_20231129.sol (hunk 3 above)
Comment (lucas-manuel) [reply src/proposals/20230913/SparkEthereum_20231129.sol:12]: Updated

File: src/proposals/20230913/SparkEthereum_20231129.sol (hunk 2 above)
Comment (lucas-manuel) [reply src/proposals/20230913/SparkEthereum_20231129.sol:10]: Updated

File: diffs/20230913-Ethereum-0xC13e21B648A5Ee794902342038FF3aDAB66BE987-pre_20230913-Ethereum-0xC13e21B648A5Ee794902342038FF3aDAB66BE987-post.md (hunk 1 above)
Comment (lucas-manuel) [reply diffs/20230913-Ethereum-0xC13e21B648A5Ee794902342038FF3aDAB66BE987-pre_20230913-Ethereum-0xC13e21B648A5Ee794902342038FF3aDAB66BE987-post.md:1]: Updated

File: src/proposals/20230913/SparkEthereum_20231129.sol (hunk 2 above)
Comment (lucas-manuel) [reply src/proposals/20230913/SparkEthereum_20231129.sol:10]: Updated

File: src/proposals/20230913/SparkEthereum_20231129.sol (hunk 3 above)
Comment (lucas-manuel) [reply src/proposals/20230913/SparkEthereum_20231129.sol:12]: Updated

File: src/proposals/20230913/SparkEthereum_20231129.t.sol (hunk 4 above)
Comment (lucas-manuel) [reply src/proposals/20230913/SparkEthereum_20231129.t.sol:151]: Updated

**Review by hexonaut — DISMISSED** [review]
Good to deploy.

**Review by hexonaut — APPROVED** [review]
LGTM

File: src/proposals/20231129/SparkEthereum_20231129.t.sol [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+        assertEq(variableBorrowRate2, 0.053790164207174267760128000e27);
+
+        // Supply rate change is slightly less than supplySpread update because of utilization
+        assertEq(variableBorrowRate2 - variableBorrowRate1, 0);
+        assertEq(supplyRate2 - supplyRate1,                 0.004747966598530556300083531e27);
+
+        // Make sure that diff is the same as start of test
+        assertApproxEqAbs(_getAssetLiabilityDiff(DAI), startDiff, 1);
+
+        // Simulate 100 days of protocol activity
```
Comment (barrutko) [inline src/proposals/20231129/SparkEthereum_20231129.t.sol:132]: Wouldn't it be better to simulate 365 days? This should give more intuitively interpretable results for anyone checking the numbers without checking the code, or performing exact calculations.

**Review by barrutko — APPROVED** [review]
- Generated diff looks good
- Comments & descriptions in the spell look good.
- Addresses in the spell are correct
- Payload address in the test is correct
- Code verified on etherscan matches the code in the repo
- All addresses and values in the tests are correct
- Spell specifics test looks good
- Asset liability test looks good
Two nits, other than that LGTM

File: src/interfaces/IERC20.sol [hunk 6]
```diff
@@ -32,6 +32,11 @@ interface IERC20 {
      */
     function balanceOf(address account) external view returns (uint256);
 
+    /**
+     * @dev Returns the token decimals.
+     */
+    function decimals() external view returns (uint8);
```
[the remark this answers is not in the document]
Comment (lucas-manuel) [reply src/interfaces/IERC20.sol:38]: You're right not needed anymore, removed.

File: src/proposals/20231129/SparkEthereum_20231129.t.sol (hunk 5 above)
Comment (lucas-manuel) [reply src/proposals/20231129/SparkEthereum_20231129.t.sol]: Updated to a year

**Review by hexonaut — APPROVED** [review]
LGTM

**Review by barrutko — APPROVED** [review]
LGTM

**Review by hexonaut — APPROVED** [review]

**Pull request merged**